gtk4/src/subclass/font_chooser.rs

```diff
@@ -0,0 +1,375 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::subclass::prelude::*;
+use crate::FontChooser;
+use glib::translate::*;
+use glib::{Cast, GString, IsA, ObjectExt, Quark};
+use once_cell::sync::Lazy;
+use pango::{FontFace, FontFamily, FontMap};
+
+#[derive(Debug)]
+pub struct FilterCallback {
+    filter_func: ffi::GtkFontFilterFunc,
+    user_data: glib::ffi::gpointer,
+    destroy_notify: glib::ffi::GDestroyNotify,
+}
+
+impl FilterCallback {
+    pub fn call(&self, font_family: &FontFamily, font_face: &FontFace) -> bool {
+        unsafe {
+            if let Some(filter_func) = self.filter_func {
+                from_glib(filter_func(
+                    font_family.to_glib_none().0,
+                    font_face.to_glib_none().0,
+                    self.user_data,
+                ))
+            } else {
+                // show the font if the filter_func was not set
+                true
+            }
+        }
+    }
+}
+
+impl Drop for FilterCallback {
+    fn drop(&mut self) {
+        unsafe {
+            if let Some(destroy_notify) = self.destroy_notify {
+                destroy_notify(self.user_data)
+            }
+        }
+    }
+}
+
+pub trait FontChooserImpl: ObjectImpl {
+    fn font_family(&self, font_chooser: &Self::Type) -> Option<FontFamily> {
+        self.parent_font_family(font_chooser)
+    }
+
+    fn font_face(&self, font_chooser: &Self::Type) -> Option<FontFace> {
+        self.parent_font_face(font_chooser)
+    }
+
+    fn font_size(&self, font_chooser: &Self::Type) -> i32 {
+        self.parent_font_size(font_chooser)
+    }
+
+    fn set_filter_func(&self, font_chooser: &Self::Type, callback: Option<&FilterCallback>) {
```

Comment:
```suggestion
    fn set_filter_func(&self, font_chooser: &Self::Type, callback: Option<FilterCallback>) {
```

Somehow the impl needs to store it, which is not possible if you just pass around a reference here.